snapshot-labs/snapshot
# page in events

### Issues
#3529

Fixes #

### Changes 
*_(Briefly describe the changes made in this PR)_*

1. this adds event paging logic for the uma module, to prevent block out of range errors and allow finding events down to block 0.  this code was adapted from https://github.com/UMAprotocol/protocol/blob/master/packages/sdk/src/oracle/utils.ts#L200

### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*

1. need to test using vercel deployment on mainnet against and open across vote that is ready to settle. if it works we should see 
![image](https://user-images.githubusercontent.com/4429761/234096654-7272ea29-551e-4496-b6ae-69a73b2b256f.png)


### To-Do
*_(List any outstanding tasks be addressed before or after this PR is merged)_*



### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
